### PR TITLE
frontend/feat: implemented metro cancellation flow

### PR DESCRIPTION
**Pull request by VD321**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Change Log:
1> Added Cancellation Flow for Metro Tickets.
2> Added City based changes for Metro Flow.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-customer/src/Flow.purs [hunk 1]
```diff
@@ -2938,8 +2938,9 @@ metroTicketBookingFlow = do
   metroStationsList <- lift $ lift $ getMetroStationsObjFromLocal ""
   let diffSec = runFn2 differenceBetweenTwoUTCInMinutes (getCurrentUTC "") metroStationsList.lastUpdatedAt
       metroStationValidTill  = config.metroTicketingConfig.metroStationTtl
+  let currentCity = getValueToLocalStore CUSTOMER_LOCATION
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Flow.purs:2941]: @VD321 Logic to cache needs to be refactored for different city/station usecase

Comment (VD321) [reply Frontend/ui-customer/src/Flow.purs:2939]: Done

File: Frontend/android-native/app/src/driver/jatriSaathiDriver/assets/juspay/juspay_assets.json [hunk 2]
```diff
@@ -236,6 +236,10 @@
       "ny_ic_chennai_metro" : true,
       "ny_ic_chennai_metro_map" : true,
       "ny_ic_chevron_arrow_right_white" : true,
+      "ny_ic_chennai_metro_banner" : true,
```
Comment (kavya-shree-s) [inline Frontend/android-native/app/src/driver/jatriSaathiDriver/assets/juspay/juspay_assets.json:239]: @VD321 This assets are customer app related , why is it added here in driver

File: Frontend/ui-customer/src/Helpers/Utils.purs [hunk 3]
```diff
@@ -910,3 +907,42 @@ findSpecialPickupZone stringGeoJson gates lat lon =
...
+  , termsAndConditions :: Array String
+  , errorPopupTitle :: String
+  , operationalHours :: 
+    { startTime :: String
+    , endTime :: String
+    }
+  , showCancelButton :: Boolean
+}
+
+getMetroConfigFromCity :: City -> CityMetroConfig
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Helpers/Utils.purs:928]: @VD321 Where do we use Metro times from this config?

Comment (VD321) [reply Frontend/ui-customer/src/Helpers/Utils.purs:938]: Shifted to v1-configurations

File: Frontend/android-native/app/src/driver/jatriSaathiDriver/assets/juspay/juspay_assets.json (hunk 2 above)
Comment (VD321) [reply Frontend/android-native/app/src/driver/jatriSaathiDriver/assets/juspay/juspay_assets.json]: Removed

File: Frontend/android-native/app/src/user/nammaYatri/assets/juspay/v1-configuration.js [hunk 4]
```diff
@@ -234,11 +234,20 @@ window.getMerchantConfig = function () {
...
+      , "cityCode" : "std:0484"      
+      , "metroStationTtl" : 10080
       , "bookingStartTime" : "04:30:00"
       , "bookingEndTime" : "22:30:00"
-    }
+      },
+      { "cityName" : "chennai"
+      , "cityCode" : "std:040"  
+      , "metroStationTtl" : 10080
+      , "bookingStartTime" : "05:00:00"
```
[the remark this answers is not in the document]
Comment (VD321) [reply Frontend/android-native/app/src/user/nammaYatri/assets/juspay/v1-configuration.js]: Fixed

**Review by kavya-shree-s — APPROVED** [review]

**Pull request merged**